tests/integration/test_case.go

```diff
@@ -360,6 +360,43 @@ type UpdateDoc struct {
 	SkipLocalUpdateEvent bool
 }
 
+// UpdateWithFilter will update the set of documents that match the given filter.
+type UpdateWithFilter struct {
```

Comment:
thought: I'm not sure this is a scaleable solution to Collection API testing and we will likely want to rework this later.  I do (currently, after a quick scan) think this is a good step however, as the (old/deleted) collection testing stuff was pretty horrible and this is a surprisingly quick way of killing it without having to worry too much about the bigger long-term testing problem (and perhaps long-term we will end up wanting one action type per API anyway, if we organise it will enough it might be the best solution).

Reply:
I'd be happy to discuss how we could simplify the collection API methods. There's a few methods that could be merged pretty easily and still keep the functionality.